### DISCUSSION TITLE
How to plot radiation pattern in polar plot with selected variations

### DISCUSSION THREAD

**Discussion by FH-DESIGN** [ansys/pyaedt#3279] [Q&A]
I try to plot the radiation pattern of a dipole antenna, but I found that I could not select the angles I want in the variations while all the angles (All phi-s) are plotted in the report. The code section is shown below:

`expressions_rad_pattern=["dB(RealizedGainTheta)"]`
`variations=hfss.available_variations.nominal_w_values_dict`
`variations["Freq"]=["3GHz"]`
`variations["Theta"]=["All"]`
`variations["Phi"]=["0deg","90deg"]`
`post.create_report(`
`expressions=expressions_rad_pattern,` 
`domain="Sweep",`
`primary_sweep_variable="Theta",`
`variations=variations,`
`report_category="Far Fields",`
`plot_type="Radiation Pattern")`

**Comment by Samuelopez-ansys** [comment]
Hi @FH-DESIGN ,

By default for radiation pattern, the secondary_sweep_variable is Phi, you need to switch to Freq:

```
hfss.post.create_report(
    "db(GainTotal)",
    hfss.nominal_adaptive,
    variations=variations,
    primary_sweep_variable="Theta",
    **secondary_sweep_variable="Freq",**
    context="3D",
    report_category="Far Fields",
    plot_type="Radiation Pattern"
)
```

I prefer to use this method, because you have more control before creating the plot:
```
new_report = hfss.post.reports_by_category.far_field("db(RealizedGainTotal)", hfss.nominal_adaptive, "3D")
new_report.variations = variations
new_report.primary_sweep = "Theta"
new_report.secondary_sweep = "Freq"
new_report.create("Realized2D")
```

**Reply by FH-DESIGN** [reply]
@Samuelopez-ansys thanks for your answer, In fact I want to have a polar plot instead. But it has been solved by Matplotlib. Thanks again.
